### PR TITLE
Added a test for rest/providers/list_all.py and wrapepd said list_all.py's web.header() function in a try-catch block

### PR DESCRIPTION
**Pull request by stanislavkozlovski**
#101

### CODE REVIEW

**Comment by stanislavkozlovski** [comment]
Please check out codeclimate's issues and instruct me further

**Comment by cglewis** [comment]
note the following lines that are still not tested (21-23, 25-26):

```
/vcontrol/vcontrol/rest/providers/list_all.py             19      5    74%   21-23, 25-26
```

**Comment by cglewis** [comment]
@Enether i think the complexity issues that codeclimate are complaining about were already there, but are showing up because you're changing code in that function/class.  going to ignore that for now, but should probably be something addressed in a different issue.

**Comment by stanislavkozlovski** [comment]
@cglewis I think this new modification should get lines 21-23. Although I don't know how to cover 25-26, because they're the except clause of `with open('providers.txt', 'r') as f:`

File: vcontrol/rest/providers/providers.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+TEST : TEST
```
Comment (cglewis) [inline vcontrol/rest/providers/providers.txt]: I'd rather not have this included in the repo, but instead have the test create this file for testing, and then clean up after itself after the test runs.

**Review by cglewis — CHANGES_REQUESTED** [review]

**Comment by cglewis** [comment]
regarding the coverage, for lines 25-26, i think it can be triggered by having a malformed line in the `providers.txt`, for example something without a `:` will throw the exception due to an index out of bounds error.

**Comment by stanislavkozlovski** [comment]
Tests are fine locally, I'm not sure what's up

File: vcontrol/rest/providers/list_all.py [hunk 2]
```diff
@@ -8,15 +8,21 @@ class ListProvidersR:
...
+        try:
+            web.header('Access-Control-Allow-Origin', self.allow_origin)
+        except Exception as e:
+            print(e.message)
         try:
             providers = {}
-            if os.path.isfile('providers.txt'):
-                with open('providers.txt', 'r') as f:
+            providers_file_path = os.path.join(os.path.abspath(__file__), '../providers.txt')
+            if os.path.isfile(providers_file_path):
```
Comment (cglewis) [inline vcontrol/rest/providers/list_all.py:21]: both of the asserts that are failing show that the output of `GET()` is `{}` which means that it's passing through this if statement and just returning `providers` as `{}`, so I think that means that the path doesn't exist on the test framework.  somehow the way you're creating it with relative paths is not the same on circleci as how you're running it locally.

**Review by cglewis — APPROVED** [review]

**Comment by cglewis** [comment]
thanks!

**Pull request merged**